src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/SymbolFactory.cs

```diff
@@ -46,27 +53,48 @@ internal static IList<ElementContainer<IAssemblySymbol>> GetElementContainersFro
+        private static CSharpCompilation CreateCSharpCompilationFromSyntax(string syntax, string name, bool enableNullable)
         {
-            CSharpCompilation compilation = CreateCSharpCompilation(name, enableNullable, includeDefaultReferences);
-            return compilation.AddSyntaxTrees(CSharpSyntaxTree.ParseText(syntax));
+            CSharpCompilation compilation = CreateCSharpCompilation(name, enableNullable);
+            return compilation.AddSyntaxTrees(GetSyntaxTree(syntax));
         }
 
-        private static CSharpCompilation CreateCSharpCompilationFromSyntax(IEnumerable<string> syntax, string name, bool enableNullable, bool includeDefaultReferences)
+        private static CSharpCompilation CreateCSharpCompilationFromSyntax(IEnumerable<string> syntax, string name, bool enableNullable)
         {
-            CSharpCompilation compilation = CreateCSharpCompilation(name, enableNullable, includeDefaultReferences);
-            IEnumerable<SyntaxTree> syntaxTrees = syntax.Select(s => CSharpSyntaxTree.ParseText(s));
+            CSharpCompilation compilation = CreateCSharpCompilation(name, enableNullable);
+            IEnumerable<SyntaxTree> syntaxTrees = syntax.Select(s => GetSyntaxTree(s));
             return compilation.AddSyntaxTrees(syntaxTrees);
         }
 
-        private static CSharpCompilation CreateCSharpCompilation(string name, bool enableNullable, bool includeDefaultReferences)
+        private static SyntaxTree GetSyntaxTree(string syntax)
+        {
+            return CSharpSyntaxTree.ParseText(syntax, ParseOptions);
+        }
+
+        private static CSharpCompilation CreateCSharpCompilation(string name, bool enableNullable)
         {
             var compilationOptions = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary,
                                                                   nullableContextOptions: enableNullable ? NullableContextOptions.Enable : NullableContextOptions.Disable);
 
-            return CSharpCompilation.Create(name, options: compilationOptions, references: includeDefaultReferences ? DefaultReferences : null);
+            // Suppress diagnostics that we don't care about in the tests.
+            compilationOptions = compilationOptions.WithSpecificDiagnosticOptions(DiagnosticOptions);
+            return CSharpCompilation.Create(name, options: compilationOptions, references: DefaultReferences);
         }
 
+        private static CSharpParseOptions ParseOptions { get; } = new(preprocessorSymbols:
+#if NETFRAMEWORK
+                new string[] { "NETFRAMEWORK" }
+#else
+                Array.Empty<string>()
+#endif
+        ); 
+
+        private static IEnumerable<KeyValuePair<string, ReportDiagnostic>> DiagnosticOptions { get; } = new[]
+        {
+            // Suppress warning for unused fields.
+            new KeyValuePair<string, ReportDiagnostic>("CS0067", ReportDiagnostic.Suppress)
```

Comment:
Where do we need this?  I wouldn't expect the compiler to ever complain about publicly visible unused fields.

Reply:
I messed up the comment it is not unused fields it was unused events in the declaring class. https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0067 

I'll update the comment.

Reply:
I see, that one makes sense to suppress.